endpoint.go

```diff
@@ -630,13 +630,44 @@ func (ep *endpoint) assignAddress() error {
 	if err != nil {
 		return err
 	}
-	for _, d := range n.getIPInfo() {
-		var addr *net.IPNet
-		addr, _, err = ipam.RequestAddress(d.PoolID, nil, nil)
+	err = ep.assignAddressVersion(4, ipam)
+	if err != nil {
+		return err
+	}
+	return ep.assignAddressVersion(6, ipam)
+}
+
+func (ep *endpoint) assignAddressVersion(ipVer int, ipam ipamapi.Ipam) error {
+	var (
+		poolID  *string
+		address **net.IPNet
+	)
+
+	n := ep.getNetwork()
+	switch ipVer {
+	case 4:
+		poolID = &ep.iface.v4PoolID
+		address = &ep.iface.addr
```

Comment:
This is for the same reason as before. I want to avoid code duplication for V4 and V6. So I operate on the pointer to the variables which will contain either v4 or v6 info. So the business logic loop is just the same.